iceoryx2/src/node/mod.rs

```diff
@@ -0,0 +1,701 @@
+            Ok(_) => (),
+            Err(NamedConceptRemoveError::InsufficientPermissions) => {
+                fail!(from origin, with NodeCleanupFailure::InsufficientPermissions,
+                    "{} {} due to insufficient permissions.", msg, entry);
+            }
+            Err(NamedConceptRemoveError::InternalError) => {
+                fail!(from origin, with NodeCleanupFailure::InsufficientPermissions,
+                    "{} {} due to an internal failure.", msg, entry);
+            }
+        }
+    }
+
+    Ok(())
+}
+
+fn remove_node_details_directory<Service: service::Service>(
+    config: &Config,
+    monitor_name: &FileName,
+) -> Result<(), NodeCleanupFailure> {
+    let origin = format!(
+        "remove_node_details_directory({:?}, {:?})",
+        config, monitor_name
+    );
+    let msg = "Unable to remove node details directory";
+    let path = node_details_path(config, monitor_name);
+    match <Service::StaticStorage as NamedConceptMgmt>::remove_path_hint(&path) {
+        Ok(()) => Ok(()),
+        Err(NamedConceptPathHintRemoveError::InsufficientPermissions) => {
+            fail!(from origin, with NodeCleanupFailure::InsufficientPermissions,
+                "{} due to insufficient permissions.", msg);
+        }
+        Err(NamedConceptPathHintRemoveError::InternalError) => {
+            fail!(from origin, with NodeCleanupFailure::InternalError,
+                "{} due to an internal error.", msg);
+        }
+    }
+}
+
+fn remove_node<Service: service::Service>(
+    id: UniqueSystemId,
+    details: &NodeDetails,
+) -> Result<bool, NodeCleanupFailure> {
+    let origin = format!(
+        "remove_node<{}>({:?})",
+        core::any::type_name::<Service>(),
+        id
+    );
+    let msg = "Unable to remove node resources";
+    let monitor_name = fatal_panic!(from origin, when FileName::new(id.value().to_string().as_bytes()),
+                                "This should never happen! {msg} since the UniqueSystemId is not a valid file name.");
+
+    let details_config = node_details_config::<Service>(&details.config, &monitor_name);
+    let detail_storages = acquire_all_node_detail_storages::<Service>(&origin, &details_config)?;
+    remove_detail_storages::<Service>(&origin, detail_storages, &details_config)?;
+    remove_node_details_directory::<Service>(details.config(), &monitor_name)?;
+
+    Ok(true)
+}
+
+/// The [`Node`] is the entry point to the whole iceoryx2 infrastructure and owns all entities.
+/// As soon as a process crashes other processes can detect dead [`Node`]s via [`Node::list()`]
+/// and clean up the stale resources - the entities that
+/// were created via the [`Node`].
+///
+/// Can be created via the [`NodeBuilder`].
+#[derive(Debug)]
+pub struct Node<Service: service::Service> {
+    id: UniqueSystemId,
+    details: NodeDetails,
+    token: ManuallyDrop<<Service::Monitoring as Monitoring>::Token>,
```

Comment:
I think I gave a bad food for thought. `token` alone is a bit nondescriptive. What do you think about `monitoring_token`?

Reply:
Would you be fine if I rename this in the follow up PR, where I touch it either way. I would like to avoid merge conflicts.